vcd/resource_vcd_rde_interface_behavior.go

```diff
@@ -52,6 +74,30 @@ func resourceVcdRdeInterfaceBehavior() *schema.Resource {
 	}
 }
 
+// hasBehaviorExecutionChanged tells Terraform whether the Behavior execution in HCL configuration has changed compared
+// to what it got from VCD, taking into account that VCD does not return fields that were created with "_internal_" or "_secure_"
+// prefix. So we must ignore those.
+func hasBehaviorExecutionChanged(_, oldValue, newValue string, d *schema.ResourceData) bool {
+	var unmarshaledOldJson, unmarshaledNewJson map[string]interface{}
+	err := json.Unmarshal([]byte(oldValue), &unmarshaledOldJson)
+	if err != nil {
+		util.Logger.Printf("[ERROR] Could not unmarshal old value JSON: %s", oldValue)
+		return false
+	}
+	err = json.Unmarshal([]byte(newValue), &unmarshaledNewJson)
+	if err != nil {
+		util.Logger.Printf("[ERROR] Could not unmarshal new value JSON: %s", newValue)
+		return false
+	}
```

Comment:
Thought - would a failed unmarshal really mean `false` (no changes)

Reply:
Yes, this got me for a second. I think the problem is the function name, which is telling the opposite of what it is really doing. I have changed it to `isBehaviorExecutionEqual` which is now more accurate.

When `hasBehaviorExecutionChanged` returns `false`, the diff is not suppressed. So yeah, it should be the other way around, hence the name change.

Now, if `isBehaviorExecutionEqual` returns `false`, it will mean that the diff is not suppressed, which makes more sense semantically.